dandiapi/api/views/asset.py

```diff
@@ -571,48 +507,29 @@ def paths(self, *args, **kwargs):
         query_serializer = AssetPathsQueryParameterSerializer(data=self.request.query_params)
         query_serializer.is_valid(raise_exception=True)
 
-        path_prefix: str = query_serializer.validated_data['path_prefix']
-        qs: QuerySet[Asset] = (
-            self.get_queryset()
-            .select_related('blob', 'embargoed_blob', 'zarr')
-            .filter(path__startswith=path_prefix)
-            .order_by('path')
+        # Permission check
+        self.raise_if_unauthorized()
+
+        # Fetch version
+        version = get_object_or_404(
+            Version,
+            dandiset__pk=versions__dandiset__pk,
+            version=versions__version,
         )
 
-        folders: dict[str, dict] = {}
-        files: dict[str, Asset] = {}
-
-        for asset in qs.iterator():
-            # Get the remainder of the path after path_prefix
-            base_path: str = asset.path[len(path_prefix) :].strip('/')
-
-            # Since we stripped slashes, any remaining slashes indicate a folder
-            folder_index = base_path.find('/')
-            is_folder = folder_index >= 0
-
-            if not is_folder:
-                # Ensure base_path is entire filename
-                base_path = asset.path[asset.path.rfind('/') + 1 :]
-                files[base_path] = asset
-            else:
-                base_path = base_path[:folder_index]
-                entry = folders.get(base_path)
-                if entry is None:
-                    folders[base_path] = {
-                        'size': asset.size,
-                        'num_files': 1,
-                        'created': asset.created,
-                        'modified': asset.modified,
-                    }
-                else:
-                    entry['size'] += asset.size
-                    entry['num_files'] += 1
-                    entry['created'] = min(entry['created'], asset.created)  # earliest
-                    entry['modified'] = max(entry['modified'], asset.modified)  # latest
-
-        items = folders
-        items.update(files)
-        resp = self._paginate_asset_paths(items)
-        return Response(resp)
+        # Fetch child paths
+        path: str = query_serializer.validated_data['path_prefix']
+        children_paths = search_asset_paths(path, version)
+        if children_paths is None:
+            raise serializers.ValidationError('Specified path not found.')
```

Comment:
[`NotFound`](https://www.django-rest-framework.org/api-guide/exceptions/#notfound) might be a more appropriate exception to raise instead of a serializer `ValidationError`; the values provided to the serializer might be completely "valid" in that they pass validation, but still point to path that doesn't exist. Plus, it raises a 404 instead of a 400.
```suggestion
            raise NotFound('Specified path not found.')
```

Reply:
[`2bbd583` (#1312)](https://github.com/dandi/dandi-archive/pull/1312/commits/2bbd583b504ab4adadf5632526fe36f6bc79faee)